subtests/docker_cli/systemd/systemd.py

```diff
@@ -5,58 +5,51 @@
 
 Operational Summary
 ---------------------
-This is to test container as systemd service using a predefined unit file
-(``p4321.service``) and a ``script p4321-server.py``. An image is built with
-this script added. Systemd starts a container using this image & the container
-writes current time to port ``4321``. From host, the socket is read and the
-value is checked to see if it is a digit. Finally, container is stopped
-through systemd and system is cleaned up.
+This is to test container as systemd service using a predefined unit file.
+Pulling, building, and running are performed.
 
 """
 
-import shutil
-import time
-import socket
 from os.path import exists
+from os.path import join
 from os import unlink
 from autotest.client import utils
-from autotest.client.shared.utils import is_port_free
-from dockertest.subtest import Subtest
-from dockertest.dockercmd import DockerCmd
+from dockertest.subtest import SubSubtest
+from dockertest.subtest import SubSubtestCaller
 from dockertest.images import DockerImages
-from dockertest.output import mustpass
-from dockertest.xceptions import DockerTestError
 
 
-class systemd(Subtest):
+class systemd(SubSubtestCaller):
+    pass
+
+
+class systemd_base(SubSubtest):
 
     def initialize(self):
-        super(systemd, self).initialize()
-        unit_file_srcpath = '{}{}'.format(self.bindir, '/p4321.service')
-        shutil.copyfile(unit_file_srcpath, self.config['sysd_unitf_dest'])
-        dkrcmd = DockerCmd(self, 'build', ['--force-rm -t',
-                                           self.config['name'], self.bindir])
-        mustpass(dkrcmd.execute())
+        super(systemd_base, self).initialize()
+        self.sub_stuff['bindir'] = self.parent_subtest.bindir
+        self.sub_stuff['unit_dest'] = join("/etc/systemd/system",
+                                           self.config['sysd_unit_file'])
+        self.sub_stuff['unit_src'] = join(self.sub_stuff['bindir'],
+                                          self.config['sysd_unit_file'])
+        self.sub_stuff['var_to_edit'] = {'NAME=': self.config['image_name'],
+                                         'DOCKERFILEDIR=':
+                                         self.sub_stuff['bindir']}
+        self.write_unitfile(self.sub_stuff['var_to_edit'])
 
     def run_once(self):
-        host = 'localhost'
-        port = 4321
         self.sysd_action('daemon-reload')
-        self.sysd_action('start', self.config['sysd_unit_file'])
-        utils.wait_for(lambda: not is_port_free(port, host), 10)
-        time_from_socket = self.read_socket(host, port)
-        self.failif(not time_from_socket.isdigit(),
-                    "Data received from container is non-numeric: '%s'"
-                    % time_from_socket)
+        for act in 'start', 'status':
+            self.sysd_action(act, self.config['sysd_unit_file'])
 
```

Comment:
I'm not seeing a `postprocess()`, and that makes me slightly uncomfortable. The systemd start & status (in `run_once`) seem to catch every error I can throw at them, but I can't help thinking that there should be some more error & status checking. I can't define it exactly so I'll keep pondering it today.